src/components/Application.tsx

```diff
@@ -401,10 +402,15 @@ export class Application extends React.Component<Props, State> {
       jobs: this.state.jobs.$filter(j => j.id !== id),
     })
     if (this.state.route.jobIds.includes(id)) {
```

Comment:
A couple of problems with this approach:

1. This conditional branch only applies to removing the job ID from the URL if it's present in there.  As such, anything that's not explicitly related to updating the browser's URL shouldn't be in this `if` block.
2. Because of number 1, `forgetJob` won't be called if the user hasn't "navigated" to it.